src/main/java/io/qdrant/client/QdrantClient.java

```diff
@@ -2380,6 +2388,65 @@ public ListenableFuture<DeleteSnapshotResponse> deleteFullSnapshotAsync(String s
 		addLogFailureCallback(future, "Delete full snapshot");
 		return future;
 	}
+	/**
+	 * Downloads a snapshot of a collection from the specified REST API URI and
+	 * saves it to the given
+	 * output path.
+	 *
+	 * @param outPath        The path where the snapshot will be saved.
+	 * @param collectionName The name of the collection.
+	 * @param snapshotName   The name of the snapshot. If null, the latest snapshot
+	 *                       will be downloaded.
+	 * @param restApiUri     The URI of the REST API. If null, the default URI
+	 *                       "http://localhost:6333"
+	 *                       will be used.
+	 */
+	public void downloadSnapshot(
+			Path outPath,
+			String collectionName,
+			@Nullable String snapshotName,
+			@Nullable String restApiUri) throws InterruptedException, IOException, ExecutionException {
```

Comment:
It's a little awkward for the caller to provide the REST API endpoint in the method call.

WDYT about creating a `QdrantHttpClient`, passing as a ctor argument to `QdrantClient`, and delegating this method to a method on `QdrantHttpClient` ?

Reply:
This was inspired from the Rust client implementation.
https://docs.rs/qdrant-client/latest/qdrant_client/client/struct.QdrantClient.html#method.download_snapshot
Since, I don't see any further usage of the REST API except for this, an explicit HTTPClient doesn't seem necessary considering the sparse usage.

Reply:
Do you know what the rationale is for allowing a consumer to provide the scheme, host and port on the method, rather than as a constructor parameter?

I can imagine a consumer also needing to configure the API key, timeouts, maybe proxy information. Even if it were a super simple HTTP client that wrapped e.g. `CloseableHttpAsyncClient` from `org.apache.httpcomponents:httpasyncclient`, it would provide an extension point to evolve in future.

Reply:
Right. The extension would be much desirable.